From 6257a7384f60e8355780f910d14c7790831323b4 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Thu, 18 Mar 2010 14:43:23 -0300
Subject: [PATCH 04/29] monitor: convert do_memory_save() to QError

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1268923406-3785-4-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 7886
O-Subject: [PATCH 3/6] monitor: convert do_memory_save() to QError
Bugzilla: 574642
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
(cherry picked from commit c34ed28b6b9915650f5f8a6353e773a4b34afce7)
---
 monitor.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 monitor.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/monitor.c b/monitor.c
index 1a3d76d..4a4990a 100644
--- a/monitor.c
+++ b/monitor.c
@@ -1453,7 +1453,7 @@ static void do_memory_save(Monitor *mon, const QDict *qdict, QObject **ret_data)
 
     f = fopen(filename, "wb");
     if (!f) {
-        monitor_printf(mon, "could not open '%s'\n", filename);
+        qemu_error_new(QERR_OPEN_FILE_FAILED, filename);
         return;
     }
     while (size != 0) {
-- 
1.7.0.3

